pkg/virt-launcher/virtwrap/network/network.go

```diff
@@ -30,30 +30,33 @@ import (
 
 const primaryPodInterfaceName = "eth0"
 
-type VMNetworkConfigurator struct {
+type NetworkingConfigurator interface {
+	Setup() error
+}
+
+type InfraNetworkingConfigurator struct {
 	vmi          *v1.VirtualMachineInstance
 	handler      netdriver.NetworkHandler
 	cacheFactory cache.InterfaceCacheFactory
+	launcherPID  int
+	nicMaker     NicGenerator
 }
 
-func newVMNetworkConfiguratorWithHandlerAndCache(vmi *v1.VirtualMachineInstance, handler netdriver.NetworkHandler, cacheFactory cache.InterfaceCacheFactory) *VMNetworkConfigurator {
-	return &VMNetworkConfigurator{
-		vmi:          vmi,
-		handler:      handler,
-		cacheFactory: cacheFactory,
-	}
-}
-
-func NewVMNetworkConfigurator(vmi *v1.VirtualMachineInstance, cacheFactory cache.InterfaceCacheFactory) *VMNetworkConfigurator {
-	return newVMNetworkConfiguratorWithHandlerAndCache(vmi, &netdriver.NetworkUtilsHandler{}, cacheFactory)
+type NetworkingSpecGenerator struct {
```

Comment:
we don't need this to be public only the constructor need to be public.

Reply:
Well, theoretically yes, but the linter frowns upon that.

```bash
$ golint pkg/virt-launcher/virtwrap/network 
pkg/virt-launcher/virtwrap/network/network.go:72:128: exported func NewInfraNetworkingConfigurator returns unexported type network.infraNetworkingConfigurator, which can be annoying to use
pkg/virt-launcher/virtwrap/network/network.go:94:126: exported func NewNetworkingSpecGenerator returns unexported type network.networkingSpecGenerator, which can be annoying to use
```

I'd rather make the linter happy than you :)

Reply:
wow, totally unexpected, I don't like it a bit though, you may want to export just an constructo to be sure that something is initialized in a proper way, I would suggest to disable the liter also the linter explanation is not very "technical"

Reply:
When it is deprecated, we can revisit this. 

I don't plan on disabling the linter.

Reply:
So look slike we should move to https://staticcheck.io/ but not at this PR, let's keep exporting the struct.